server.js

```diff
@@ -300,89 +342,88 @@ function resolveWalkthroughLocations(locations) {
-                getWalkthroughHeader(cloned.localDir)
-                  .then(head => {
-                    let wtHeader;
-                    if (head === null) {
-                      wtHeader = null;
-                    } else {
-                      wtHeader = head.prettyName;
-                    }
-                    const walkthroughFolders = [];
-                    if (!Array.isArray(walkthroughParams.walkthroughsFolder)) {
-                      walkthroughFolders.push(walkthroughParams.walkthroughsFolder || 'walkthroughs');
-                    } else {
-                      walkthroughFolders.push(...walkthroughParams.walkthroughsFolder);
-                    }
-                    // Get the folders to import in the repository.
-                    const walkthroughInfos = walkthroughFolders.map(folder => {
-                      const walkthroughLocationInfo = Object.assign({}, WALKTHROUGH_LOCATION_DEFAULT, {
-                        type: WALKTHROUGH_LOCATION_TYPE_GIT,
-                        commitHash: log.latest.hash,
-                        commitDate: log.latest.date,
-                        remote: cloned.repoName,
-                        directory: folder,
-                        header: wtHeader
-                      });
-
-                      return Object.assign({}, locationResultTemplate, {
-                        parentId: `${repoName}-${path.basename(folder)}`,
-                        walkthroughLocationInfo,
-                        local: path.join(cloned.localDir, folder)
-                      });
-                    });
-                    resolve(walkthroughInfos);
-                  })
-                  .catch(reject);
-              });
+  const mappedLocations = locations.map(location =>
+    new Promise((resolve, reject) => {
+      const locationResultTemplate = { origin: location };
+      if (!location) {
+        return reject(new Error(`Invalid location ${location}`));
+      } else if (isPath(location)) {
+        console.log(`Importing walkthrough from path ${location}`);
+        const locationResult = Object.assign(
+          {
+            parentId: path.basename(location),
+            walkthroughLocationInfo: Object.assign({}, WALKTHROUGH_LOCATION_DEFAULT, {
+              type: WALKTHROUGH_LOCATION_TYPE_PATH,
+              directory: path.basename(location)
             })
-            .catch(reject);
-        }
+          },
+          locationResultTemplate,
+          { local: location }
+        );
+        return resolve(locationResult);
+      } else if (isGitRepo(location)) {
+        const clonePath = path.join(TMP_DIR, tmpDirPrefix);
+
+        // Need to parse out query params for walkthroughs, e.g custom directory
+        const cloneUrl = generateCloneUrlFromLocation(location);
+        const repoName = getWalkthroughRepoNameFromLocation(location);
+        const walkthroughParams = querystring.parse(url.parse(location).query);
+
```

Comment:
most of these seem like lint fixes that are finally being done so the changes seem worse than they actually are... makes it tough to discern only the changes for this specific feature. i'd rather they be handled separately for that exact reason but in this case it's fine.

Reply:
okay not sure how to undo the linting changes.

Reply:
i'd just keep them in, they've bugged me for a while now anyway - server.js is exempt from linting blockage that's why it's always managed to slip through the cracks